container/containerd/client.go

```diff
@@ -62,29 +64,37 @@ func Client(address, namespace string) (ContainerdClient, error) {
 		}
 		tryConn.Close()
 
+		connParams := grpc.ConnectParams{
+			Backoff: backoff.Config{
+				BaseDelay: baseBackoffDelay,
+				MaxDelay:  maxBackoffDelay,
+			},
+		}
 		gopts := []grpc.DialOption{
 			grpc.WithInsecure(),
-			grpc.WithDialer(dialer.Dialer),
+			grpc.WithContextDialer(dialer.ContextDialer),
 			grpc.WithBlock(),
-			grpc.WithBackoffMaxDelay(maxBackoffDelay),
-			grpc.WithTimeout(connectionTimeout),
+			grpc.WithConnectParams(connParams),
 		}
 		unary, stream := newNSInterceptors(namespace)
 		gopts = append(gopts,
 			grpc.WithUnaryInterceptor(unary),
 			grpc.WithStreamInterceptor(stream),
 		)
 
-		conn, err := grpc.Dial(dialer.DialAddress(address), gopts...)
+		ctx, cancel := context.WithTimeout(context.Background(), connectionTimeout)
+		conn, err := grpc.DialContext(ctx, dialer.DialAddress(address), gopts...)
 		if err != nil {
 			retErr = err
+			cancel()
```

Comment:
should we just `defer cancel()` after the `context.WithTimeout(...` call?

Reply:
Sounds legit.